packages/aragon-wrapper/src/index.js

```diff
@@ -222,39 +222,61 @@ export default class Aragon {
    *         The address of the proxy to get metadata from
    * @return {Promise<Object>}
    */
-  async getAppProxyValues (proxyAddress) {
+  async getProxyValues (proxyAddress) {
     // This function caches information about the AppProxy, as it is called for
     // all the apps everytime a permission changes and this data won't change
     // once it's fetched
-    const cachedValue = appProxyValuesCache[proxyAddress]
+    const cachedValue = proxyValuesCache[proxyAddress]
     if (cachedValue && cachedValue.kernelAddress && cachedValue.appId && cachedValue.codeAddress) {
       return cachedValue
     }
 
-    const appProxy = makeProxy(proxyAddress, 'AppProxy', this.web3, this.kernelProxy.initializationBlock)
-
-    const appProxyValues = await Promise.all([
-      appProxy.call('kernel').catch(() => null),
-      appProxy.call('appId').catch(() => null),
-      appProxy
-        .call('implementation')
-        .catch(() => appProxy
-          // Fallback to old non-ERC897 proxy implementation
-          .call('getCode')
-          .catch(() => null)
-        ),
-      appProxy.call('isForwarder').catch(() => false)
-    ]).then((values) => ({
-      proxyAddress,
-      kernelAddress: values[0],
-      appId: values[1],
-      codeAddress: values[2],
-      isForwarder: values[3]
-    }))
-
-    appProxyValuesCache[proxyAddress] = appProxyValues
+    let proxyValues
+
+    if (addressesEqual(proxyAddress, this.kernelProxy.address)) {
+      const kernelProxy = makeProxy(proxyAddress, 'AppProxy', this.web3, this.kernelProxy.initializationBlock)
```

Comment:
Shouldn't it have the `KernelProxy` interface?

Reply:
That is code that @sohkai added, but I'm pretty sure it was b/c there is no `KernelProxy` abi in `aragon-wrapper/abi/aragon` right now

Reply:
We could add `KernelProxy`, or just a generic `ERCProxy` abi instead. I mainly used `AppProxy` out of laziness (we only needed it for `implementation`, which is available in `AppProxy`).